core/src/main/java/com/linecorp/armeria/common/Scheme.java

```diff
@@ -36,19 +36,19 @@
  * For example:
  * </p>
  * <ul>
- * <li>{@code "tbinary+https"} - {@link SerializationFormat#THRIFT_BINARY} and
- *                               {@link SessionProtocol#HTTPS}</li>
- * <li>{@code "tcompact+h2c"} - {@link SerializationFormat#THRIFT_COMPACT} and
- *                              {@link SessionProtocol#H2C}</li>
+ * <li>{@code "tbinary+https"}</li>
+ * <li>{@code "tcompact+h2c"}</li>
+ * <li>{@code "none+http"}</li>
  * </ul>
  */
 public final class Scheme implements Comparable<Scheme> {
 
     private static final Map<String, Scheme> SCHEMES;
 
     static {
-        // Pre-populate all known scheme combos.
-        final Map<String, Scheme> schemes = new HashMap<>();
+        // Pre-populate all possible scheme combos.
+        // XXX(trustin): We may want to make this lazy once we support more than HTTP and Thrift.
```

Comment:
I was rather thinking about the case where we support a lot of session protocols and serialization formats which will result in numSessProto * numSerFmt / 2 instances of scheme where vast majority of them are unused or impossible.